casper/src/main/scala/coop/rchain/casper/EstimatorHelper.scala

```diff
@@ -115,14 +116,13 @@ object EstimatorHelper {
       comm.consume.channelsHashes ++ comm.produces.map(_.channelsHash)
     }.toSet
 
-  private[this] def extractBlockEvents[F[_]: Monad: BlockStore](
+  private[this] def extractBlockEvents[F[_]: Sync: BlockStore](
       blockAncestorsMeta: List[BlockMetadata]
   ): F[BlockEvents] =
     for {
-      maybeAncestors <- blockAncestorsMeta.traverse(
-                         blockAncestorMeta => BlockStore[F].get(blockAncestorMeta.blockHash)
-                       )
-      ancestors = maybeAncestors.flatten
+      ancestors <- blockAncestorsMeta.traverse(
+                    blockAncestorMeta => BlockStore[F].getUnsafe(blockAncestorMeta.blockHash)
+                  )
```

Comment:
This code can not throw errors which should be what we want and not to skip ancestors if not in the store.

Reply:
Why here you changed Monad to Sync?

Reply:
Because of use of `Sync.raiseError` inside `BlockStore[F].getUnsafe`.